main.go

```diff
@@ -0,0 +1,193 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"runtime"
+
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/fileutil"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-tools/go-steputils/stepconf"
+	semver "github.com/hashicorp/go-version"
+)
+
+// Config model
+type Config struct {
+	Workdir    string `env:"workdir"`
+	Command    string `env:"command,required"`
+	NpmVersion string `env:"npm_version"`
+}
+
+func getNpmVersionFromPackageJSON() (string, error) {
+	jsonStr, err := fileutil.ReadStringFromFile("package.json")
+	if err != nil {
+		return "", fmt.Errorf("package.json file read error: %s", err)
+	}
+
+	var ver string
+	if ver, err = extractNpmVersion(jsonStr); err != nil {
+		return "", fmt.Errorf("package json content error: %s", err)
+	}
+	return ver, nil
+}
+
+func extractNpmVersion(jsonStr string) (string, error) {
```

Comment:
this function cares about package.json file, i would call `jsonStr` as `packageJSON`